pkg/yurttunnel/agent/cmd.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agent
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/certificate"
+	"k8s.io/klog"
+
+	"github.com/alibaba/openyurt/pkg/yurttunnel/pki"
+	"github.com/alibaba/openyurt/pkg/yurttunnel/pki/certmanager"
+)
+
+// NewYurttunnelAgentCommand creates a new yurttunnel-agent command
+func NewYurttunnelAgentCommand(stopCh <-chan struct{}) *cobra.Command {
+	o := NewYurttunnelAgentOptions()
+
+	cmd := &cobra.Command{
+		Short: "Launch edge-tunnel-agent",
+		Long:  "Launch edge-tunnel-agent",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.validate(); err != nil {
+				return err
+			}
+			if err := o.run(stopCh); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&o.nodeName, "node-name", o.nodeName,
+		"The name of the edge node.")
+	flags.StringVar(&o.serverAddr, "server-addr", o.serverAddr,
+		"The address of yurttunnel-server")
+	flags.StringVar(&o.kubeConfig, "kube-config", o.kubeConfig,
+		"Path to the kubeconfig file")
+	flags.StringVar(&o.caFile, "ca-file", o.caFile,
+		"The CA file for securing the connection to the yurttunnel-server ")
+
+	return cmd
+}
+
+// YurttunnelAgentOptions has the information that required by the
+// yurttunel-agent
+type YurttunnelAgentOptions struct {
+	nodeName   string
+	serverAddr string
+	kubeConfig string
+	caFile     string
+}
+
+// NewYurttunnelAgentOptions creates a new YurtNewYurttunnelAgentOptions
+func NewYurttunnelAgentOptions() *YurttunnelAgentOptions {
+	o := &YurttunnelAgentOptions{
+		kubeConfig: "/etc/kubernetes/kubelet.conf",
+		caFile:     "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+	}
+	return o
+}
+
+// validate validates the YurttunnelServerOptions
+func (o *YurttunnelAgentOptions) validate() error {
+	if o.kubeConfig == "" {
+		if _, err := os.Stat(o.kubeConfig); os.IsNotExist(err) {
+			return fmt.Errorf("%s doesn't exist", o.kubeConfig)
+		}
+		if o.caFile == "" {
+			return errors.New("either '--kube-config' or " +
+				"'--ca-file' need to be set")
+		}
+		if _, err := os.Stat(o.caFile); os.IsNotExist(err) {
```

Comment:
when ca file is not exist, would you get is from kube-apiserver?

Reply:
You mean from kubeconfig?